File: app/schemas/fear_greed.py
"""Fear & Greed Index Pydantic schemas."""
from typing import List, Optional

from pydantic import Field

from app.schemas.base import BaseResponse


class FearGreedPoint(BaseResponse):
    """历史数据中的单个数据点。"""

    date: str = Field(description="日期，格式 YYYY-MM-DD")
    score: float = Field(ge=0, le=100, description="恐慌贪婪指数分值 0–100")
    rating: str = Field(description="情绪标签，如 Extreme Fear / Fear / Neutral / Greed / Extreme Greed")


class FearGreedSnapshot(BaseResponse):
    """特定时间点的快照（当前/前一周/前一月/前一年/历史极值）。"""

    score: float = Field(ge=0, le=100)
    rating: str
    date: Optional[str] = Field(None, description="仅当前值、历史最高/最低时携带")


class FearGreedResponse(BaseResponse):
    """GET /api/v1/fear-greed 完整响应。"""

    current: FearGreedSnapshot
    previous_week: FearGreedSnapshot
    previous_month: FearGreedSnapshot
    previous_year: FearGreedSnapshot
    history_low: FearGreedSnapshot
    history_high: FearGreedSnapshot
    history: List[FearGreedPoint]

File: frontend/lib/api/fear_greed.ts
// frontend/lib/api/fear_greed.ts
import apiClient from './client'

export interface FearGreedPoint {
  date: string
  score: number
  rating: string
}

export interface FearGreedSnapshot {
  score: number
  rating: string
  date?: string
}

export interface FearGreedResponse {
  request_id: string
  current: FearGreedSnapshot
  previous_week: FearGreedSnapshot
  previous_month: FearGreedSnapshot
  previous_year: FearGreedSnapshot
  history_low: FearGreedSnapshot
  history_high: FearGreedSnapshot
  history: FearGreedPoint[]
}

export async function fetchFearGreed(
  startDate?: string,
  endDate?: string
): Promise<FearGreedResponse> {
  const params: Record<string, string> = {}
  if (startDate) params.start_date = startDate
  if (endDate) params.end_date = endDate
  
  const { data } = await apiClient.get<FearGreedResponse>('/api/v1/fear-greed', { params })
  return data
}
